### PR TITLE
Alternate sending bulk telemetry over AMQP and parameterize payload convention

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: e2e/LongHaul/device/Parameters.cs [hunk 1]
```diff
@@ -34,7 +40,7 @@ internal class Parameters
             "Transport",
             Default = TransportType.Mqtt,
             Required = false,
-            HelpText = "The transport to use for the connection.")]
+            HelpText = "The transport to use for the connection (i.e. Mqtt, Amqp).")]
```
Comment (patilsnr) [inline e2e/LongHaul/device/Parameters.cs:43]: very nit but perhaps `e.g.` is more gramatically correct

File: e2e/LongHaul/device/Parameters.cs [hunk 2]
```diff
@@ -44,5 +50,33 @@ internal class Parameters
             Required = false,
             HelpText = "The protocol over which a transport (i.e., MQTT, AMQP) communicates.")]
         public IotHubClientTransportProtocol TransportProtocol { get; set; }
+
+        [Option(
+            'j',
+            "JsonSerializer",
+            Default = JsonSerializingLibrary.SystemTextJson,
+            Required = false,
+            HelpText = "Which JSON serialization library for the device app to use (i.e., SystemTextJson, NewtonsoftJson)")]
```
Comment (patilsnr) [inline e2e/LongHaul/device/Parameters.cs:59]: same here, also this `i.e.,` has a comma afterward while the previous one doesn't. I think either work but we should be consistent in theory

File: e2e/LongHaul/device/Parameters.cs (hunk 1 above)
Comment (patilsnr) [reply e2e/LongHaul/device/Parameters.cs:43]: actually, if `amqp` and `mqtt` are the only options then `i.e.` works

**Review by patilsnr — APPROVED** [review]

File: e2e/LongHaul/device/Parameters.cs (hunk 1 above)
Comment (drwill-ms) [reply e2e/LongHaul/device/Parameters.cs:43]: They are the only options.

File: e2e/LongHaul/device/Parameters.cs (hunk 2 above)
Comment (drwill-ms) [reply e2e/LongHaul/device/Parameters.cs:59]: I'll fix the missing comma

File: e2e/LongHaul/device/IotHub.cs [hunk 3]
```diff
@@ -135,23 +140,48 @@ public async Task SendTelemetryMessagesAsync(Logger logger, CancellationToken ct
...
+                    }
                     sw.Stop();
 
-                    _logger.Metric(TotalTelemetryMessagesSent, ++_totalTelemetryMessagesSent);
+                    _totalTelemetryMessagesSent += pendingMessages.Count;
+                    _logger.Metric(TotalTelemetryMessagesSent, _totalTelemetryMessagesSent);
                     _logger.Metric(TelemetryMessageDelaySeconds, sw.Elapsed.TotalSeconds);
 
-                    pendingMessage = null;
+                    sendSingle = !sendSingle;
```
Comment (brycewang-microsoft) [inline e2e/LongHaul/device/IotHub.cs:183]: why flip `sendSingle` after sending all pending messages?

**Review by brycewang-microsoft — APPROVED** [review]

File: e2e/LongHaul/device/IotHub.cs (hunk 3 above)
Comment (drwill-ms) [reply e2e/LongHaul/device/IotHub.cs:183]: To alternate between sending in bulk or single so AMQP gets both paths tested. I can add a code comment.

Comment (brycewang-microsoft) [reply e2e/LongHaul/device/IotHub.cs:183]: got it. Sure, that could help!

File: e2e/LongHaul/device/IotHub.cs [hunk 4]
```diff
@@ -101,9 +100,19 @@ public async Task InitializeAsync()
...
+                ? 10
+                : 1;
+
+            // We want to test both paths for AMQP so we'll use a boolean to alternate between bulk and single.
+            bool sendSingle = true;
+
+            var pendingMessages = new List<TelemetryMessage>(maxBulkMessages);
             bool loggedDisconnection = false;
             logger.LoggerContext.Add(OperationName, LoggingConstants.TelemetryMessage);
+            var sw = new Stopwatch();
```
Comment (drwill-ms) [inline e2e/LongHaul/device/IotHub.cs:114]: @tmahmood-microsoft here's an example of stop watch usage I'm talking about.

**Pull request merged**